### PR TITLE
Fix for Missing Logs in ASP.NET Core 6.0 Web and Razor Apps

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
## Why

ASP.NET Core Razor and several types of web applications do not produce logs when auto-instrumentation is enabled. This issue occurs exclusively on .NET 6.0.

**Reproduction Steps:**
To reproduce this issue, create a default ASP.NET Core 6.0 Razor web application. After auto-instrumenting with the 1.0.1 release, no logs will be collected.

## What

The problem originated from a specific line in the `LogBuilderExtensions` code. For the affected application types, the return type is `Microsoft.AspNetCore.WebApplicationServiceCollection`. Consequently, the following logic fails:

```
if (!(builder.Services is ServiceCollection services))
```

The fix involves check builder.Services is null in the aforementioned line. During the debugging process, I noticed that the logs did not provide enough clarity regarding this issue. To improve this, I've added a verbose log for better traceability.

## Recommendation:

Considering this issue affects all ASP.NET Core 6.0 applications and prevents any log emissions, it's crucial to **release** version **1.0.2** after merging this PR.

## Tests

Existing CI.

## Checklist

- [X] `CHANGELOG.md` is updated.
- [ ] ~~Documentation is updated.~~
- [X] New features are covered by tests.

### CODE REVIEW

**Review by pjanotti — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs [hunk 1]
```diff
@@ -31,8 +31,9 @@ public static ILoggingBuilder AddOpenTelemetryLogs(this ILoggingBuilder builder)
     {
         try
         {
-            if (!(builder.Services is ServiceCollection services))
+            if (!(builder.Services is IServiceCollection services))
```
Comment (Kielek) [inline src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs:34]: In general, you can change the condition to builder.Services == null.

ILoggingBuilder Services is always `IServiceCollection`.

Shouldn't it check some extended supported set of types?

**Review by Kielek — COMMENTED** [review]
Please include also test to the codebase to cover such scenario. 

I am fine with 1.0.2 just after this release.

File: src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs:34]: Good thought, we could check this for null. I don't recall the reason for checking ServicesCollection here.

**Comment by rajkumar-rangaraj** [comment]
> Please include also test to the codebase to cover such scenario. If possible without bootstrap and jQuery files from default razor template.
> 
> I am fine with 1.0.2 just after this is merged.

We already have tests for logs. I could add another one for the Razor app. However, I'm uncertain about its value since this test would never reproduce the issue. If there's a problem, other tests should detect it.

File: src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs]: Fixed.

**Comment by rajkumar-rangaraj** [comment]
@Kielek Added tests. We could remove it later if we find this as duplicate.

File: test/test-applications/integrations/TestApplication.Azure/Program.cs
Comment (Kielek) [inline test/test-applications/integrations/TestApplication.Azure/Program.cs:1]: I think that this files was added by mistake.
When reverted we are fine to merge.

Comment (rajkumar-rangaraj) [reply test/test-applications/integrations/TestApplication.Azure/Program.cs:1]: Thanks for spotting out. Fixed.

**Review by Kielek — APPROVED** [review]

**Pull request merged**